indigo_api/models/works.py

```diff
@@ -359,6 +350,44 @@ def as_at_date(self):
         if dates:
             return max(dates)
 
+    def all_provisions(self):
+        # a list of the element ids of the earliest PiT (in whichever language) of a work
+        # does not include Schedules as they don't have commencement dates
+        first_expression = self.expressions().first()
+        if first_expression:
+            return first_expression.all_provisions()
+        return None
+
+    def main_commencement(self):
+        main = self.commencements.filter(main=True).first()
```

Comment:
`return self.commencements.filter(main=True).first()`

Reply:
Rather cache this method, than the "alias" property. This is the one that goes to the DB.